packages/core/src/domain/telemetry/telemetry.ts

```diff
@@ -76,10 +77,17 @@ export function startTelemetry(telemetryService: TelemetryService, configuration
 
   const runtimeEnvInfo = getRuntimeEnvInfo()
   onRawTelemetryEventCollected = (rawEvent: RawTelemetryEvent) => {
-    if (telemetryEnabledPerType[rawEvent.type!]) {
+    const stringifiedEvent = jsonStringify(rawEvent)!
```

Comment:
**💭 thought:** ‏I don't love the extra serialization and memory usage (as we store a serialized version of every telemetry event forever), but I don't have a simpler/better idea.

Reply:
> we store a serialized version of every telemetry event forever

only the first `maxEventsPerPage` events though, so 15 for now.

Reply:
We could avoid to serialize if we know that we won't send the event.
Do you think it worth adding this additional code path?

Reply:
Hum, maybe not.

While having another look, I thought of a small simplification: we could replace `telemetryConfiguration.sentEventCount` with `alreadySeenEvents.size` (one less state to maintain!)